src/app/SharpRaven/RavenClient.cs

```diff
@@ -363,6 +393,19 @@ protected virtual string Send(JsonPacket packet)
             }
             catch (Exception exception)
             {
+                if (!cachePacketsWhenOffline) return HandleException(exception, requester);
+                try
+                {
+                    if (!string.IsNullOrEmpty(packet.Message)) packet.Message += "\n";
+                    packet.Message +=
+                        $"Attempted to send on {DateTime.UtcNow.ToShortDateString()} {DateTime.UtcNow.ToLongTimeString()} UTC however packet failed to send and was cached locally.";
```

Comment:
I like that you add this info, but I think `Message` should be left untouched. Can you perhaps set this message inside `extra` somewhere instead?